dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -244,26 +246,52 @@ public BinaryRecordReader(InputStream is, long streamPos, Span recBuildSpan) thr
         {
             throw new Exception("BinaryRecordReader requires provided InputStream to support mark()");
         }
+
+        startNewRecordBatchSpan();
+    }
+
+    /**
+     * Set the record batch size in KB.
+     * 
+     * @param recordBatchSizeInKB the record batch size in KB
+     */
+    public void setRecordBuildingSpanBatchSizeKB(int recordBatchSizeInKB)
+    {
+        if (recordBatchSizeInKB <= 0)
+        {
+            return;
+        }
+
+        this.recordBatchSize = recordBatchSizeInKB * 1024;
+    }
+
+    private void endRecordBatchSpan()
+    {
+        if (this.recordBatchSpan == null)
```

Comment:
is it possible for recordBatchSpan != null, but be invalid/not recording?  if so, could we skip the stop process?

Reply:
Good question, I added a check when setting the initial span in the BinaryRecordReader to null out the span if it isn't valid.